scipy/stats/_discrete_distns.py

```diff
@@ -492,18 +492,48 @@ def _entropy(self, M, n, N):
         vals = self.pmf(k, M, n, N)
         return np.sum(entr(vals), axis=0)
 
-    def _sf(self, k, M, n, N):
-        # This for loop is needed because `k` can be an array. If that's the
-        # case, the sf() method makes M, n and N arrays of the same shape. We
-        # therefore unpack all inputs args, so we can do the manual
-        # integration.
+    def _sum_pmf(self, k, M, n, N, cdf):
+        # Sum the tail of the PMF, and then subtract the sum from 1 if
+        # necessary to return the value indicated by the boolean `cdf`.
+        # If k > mode, sum the right tail.  Otherwise sum the left tail.
+        #
+        # k, M, n and N are either all scalars or all 1-d arrays.
```

Comment:
When are they all scalars? Often they are converted to arrays and then converted back to scalars at the public method level, right? (`moment` is an exception, but are there others you've found? _Shouldn't_ this always be taken care of at the public method level? )

Reply:
Calling `ppf` (which uses the default implementation in `hypergeom`) will result in several calls to `_cdf` with scalars.  The `median` and `interval` methods depend on `ppf`, so they will also result in calls of `_cdf` with scalars.